rviz_default_plugins/include/rviz_default_plugins/displays/effort/effort_display.hpp

```diff
@@ -0,0 +1,161 @@
+#include <rviz_common/message_filter_display.hpp>
+#include <rviz_common/properties/bool_property.hpp>
+#include <rviz_common/properties/int_property.hpp>
+#include <rviz_common/properties/float_property.hpp>
+#include <rviz_common/properties/string_property.hpp>
+#include <rviz_common/properties/property.hpp>
+#include <rviz_rendering/objects/effort_visual.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <std_msgs/msg/string.hpp>
+
+#include "rviz_default_plugins/visibility_control.hpp"
+
+namespace rviz_default_plugins
+{
+namespace displays
+{
+class JointInfo : public QObject
+{
+  Q_OBJECT
+
+public:
+  JointInfo(const std::string & name, rviz_common::properties::Property * parent_category);
+  ~JointInfo() override;
+
+  void setEffort(double e);
+  inline double getEffort()
+  {
+    return effort_;
+  }
+  void setMaxEffort(double m);
+  inline double getMaxEffort()
+  {
+    return max_effort_;
+  }
+  bool getEnabled() const;
+
+  rclcpp::Time last_update_;
+
+public Q_SLOTS:
+  void updateVisibility();
+
+private:
+  std::string name_;
+  double effort_, max_effort_;
+
+  rviz_common::properties::Property * category_;
+  rviz_common::properties::FloatProperty * effort_property_;
+  rviz_common::properties::FloatProperty * max_effort_property_;
+};
+
+class RVIZ_DEFAULT_PLUGINS_PUBLIC EffortDisplay
+  : public rviz_common::MessageFilterDisplay<sensor_msgs::msg::JointState>
+{
+  Q_OBJECT
+
+public:
+  EffortDisplay();
+  ~EffortDisplay() override;
+
+  // Overrides of public virtual functions from the Display class.
+  void onInitialize() override;
+  void reset() override;
+
+  void load(const rviz_common::Config & config) override;
+
+  void update(float wall_dt, float ros_dt) override;
+
+private Q_SLOTS:
+  // Helper function to apply color and alpha to all visuals.
+  void updateColorAndAlpha();
+  void updateHistoryLength();
+  void updateRobotDescription();
+  void updateTfPrefix();
+
+private:
+  JointInfo * getJointInfo(const std::string & joint);
+  JointInfo * createJoint(const std::string & joint);
+  void subscribeToRobotDescription();
+
+protected:
+  // overrides from Display
+  void onEnable() override;
+  void onDisable() override;
+
+  void load();
+  void clear();
+
+  // The object for urdf model
+  std::shared_ptr<urdf::Model> robot_model_;
+
+  std::string robot_description_;
+  std::string robot_description_topic_;
+
+private:
+  void processMessage(sensor_msgs::msg::JointState::ConstSharedPtr msg) override;
+  void topic_callback(const std_msgs::msg::String & msg);
+
+  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
+  rclcpp::executors::SingleThreadedExecutor::SharedPtr executor_;
+
+  // Storage for the list of visuals.  It is a circular buffer where
+  // data gets popped from the front (oldest) and pushed to the back (newest)
+  std::deque<std::shared_ptr<rviz_rendering::EffortVisual>> visuals_;
+
+  typedef std::map<std::string, JointInfo *> M_JointInfo;
```

Comment:
I just had a thought about this.  If we instead made this:

```suggestion
  typedef std::map<std::string, std::unique_ptr<JointInfo>> M_JointInfo;
```

Then we wouldn't need to explicitly free it in the destructor.  We'd also have to change where we allocate it to `std::make_unique`, but I think that is probably worth it.